libraries/wallet/wallet.cpp

```diff
@@ -772,8 +787,17 @@ class wallet_api_impl
                        return _keys.find( pk ) != _keys.end();
                     } );
 
+      if ( erase_existing_sigs )
+         tx.signatures.clear();
```

Comment:
Since `get_owned_required_keys` aims at code reusing, move this two lines out would somehow made this function unneeded.

It's true that modifying out params in function is not a good practice. But I have an another thought, how about passing value (instead of reference) to `get_owned_required_keys`? Are we really care about the tiny performance improvement by passing reference here?

Reply:
`get_owned_required_keys` is added in order of code reusing. If moved these two lines out, the function can be removed too